README.md

```diff
@@ -2,158 +2,225 @@
+> In our case we use a **`.env`** file
+to manage our environment variables.
+See:
+[github.com/dwyl/**learn-environment-variables**](https://git.io/JeMLg) <br />
+This allows us to securely manage our secret keys in dev
+without the risk of accidentally publishing them on Github. <br />
+When we _deploy_ our Apps, we use our service provider's
+built-in key management service to securely store Environment Variables.
+e.g:
+[Environment Variables on Heroku](https://github.com/dwyl/learn-environment-variables#environment-variables-on-heroku)
 
-``` elixir
-config :fields, Fields,
-  secret_key_base: "rVOUu+QTva+VlRJJI3wSYONRoffFQH167DfiZcegvYY/PEasjPLKIDz7wPTvTPIP"
-```
 
-## Background / Further Reading
+## 3. Apply the relevant field(s) to your schema 📝
 
+Each field can be used in place of an Ecto type when defining your schema.
 
-### How to Create a Re-useable Elixir Package and Publish to `hex.pm`
-+ https://hex.pm/docs/publish
-+ https://medium.com/kkempin/how-to-create-and-publish-hex-pm-package-elixir-90cb33e2592d
-+ https://medium.com/blackode/how-to-write-elixir-packages-and-publish-to-hex-pm-8723038ebe76
+An example for defining a "user" schema using **Fields**:
 
-### How to _Use_ the Package Locally _Before_ Publishing it to `hex.pm`
+```elixir
+schema "users" do
+  field :first_name, Fields.Name            # Length validated and encrypted
+  field :email, Fields.EmailEncrypted       # Validates email then encrypts
+  field :address, Fields.AddressEncrypted   # Trims address string then encrypts
+  field :postcode, Fields.PostcodeEncrypted # Validates postcode then encrypts
+  field :password, Fields.Password          # Hash password with argon2
 
-+ https://stackoverflow.com/questions/28185003/using-a-package-locally-with-hex-pm
+  timestamps()
+end
+```
+
+Each field is defined as an
+[Ecto type](https://hexdocs.pm/ecto/Ecto.Type.html),
+with the relevant callbacks.
+So when you call `Ecto.Changeset.cast/4`
+in your schema's changeset function,
+the field will be correctly validated.
+For example, calling cast on the `:email` field
+will ensure it is a valid format for an email address
+[RFC 5322](https://en.wikipedia.org/wiki/Email_address).
+
+When you load one of the fields into your database,
+the corresponding `dump/1` callback will be called,
+ensuring it is inserted into the database in the correct format.
+In the case of `Fields.EmailEncrypted`,
+it will encrypt the email address
+using a given encryption key
+before inserting it.
+
+Likewise, when you load a field from the database,
+the `load/1` callback will be called,
+giving you the data in the format you need.
+`Fields.EmailEncrypted` will be decrypted back to plaintext.
+This all happens 100% transparently to the developer.
+It's _like_ magic. But the kind where you can
+actually _understand_ how it works!
+(_if you're curious, read the
+[**`code`**](https://github.com/dwyl/fields/tree/master/lib)_)
+
+Each Field optionally defines an `input_type/0` function.
```

Comment:
At the moment only the `HtmlBody` and `DescriptionPlaintextUnlimited` fields define this function. We might want to add it to the other fields

Reply:
@SimonLab absolutely. We need to add this to the other fields ASAP. 👍 
Would you mind creating an issue with a checklist for all the remaining fields that don't yet have it?